File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.swift
import Foundation

func solution(_ n:Int) -> Int {
    var sum = 0
    if n % 2 == 0 {
        for i in 2 ... n {
            if i % 2 == 0 {
                sum += i*i
            }
        }
    }else{
        for i in 1 ... n {
            if i % 2 == 1 {
                sum += i
            }
        }
    }
    return sum
}

File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.js
function solution(n) {
    let result = 0
    if (n%2 === 0) {
        for(n; n>0; n = n-2) {
            result += n*n
        }
    } else {
        for (n; n>0; n = n-2) {
            result += n
        }
    }
    return result;
}